TheJumpCloud/support
# Sa 2078 codeartifact updates

## Issues
* [SA-2078](https://jumpcloud.atlassian.net/browse/SA-2078) - Update-JCModule Supports CodeArtifact Repository

## What does this solve?
This set of changes to the Update-JCModule function includes support for CodeArtifact as a source repository. The JumpCloud module can be updated from it's public latest version in PSGallery to a development version in CodeArtifact with the `Update-JCModule -Repository:('codeartifact') -RepositoryCredentials:($RepositoryCredentials)` function parameters.

## Is there anything particularly tricky?
This change should not affect customers, without credentials to access the code artifact repository in AWS, customers should not be able to access this resource. 

## How should this be tested?
There are three scenarios to test:

- Older module installed from PSGallery can be updated to latest module in PSGallery with `Update-JCModule`
- Module installed from PSGallery can be updated to latest module in CodeArtifact with `Update-JCModule -Repository:('codeartifact') -RepositoryCredentials:($RepositoryCredentials)`
- After a release is published, module installed from codeartifact can be updated to the latest version in PSGallery with the `Update-JCModule` function

Running the function for the first time should prompt for users to install a later version of PowerShellGet. This set of changes was written for PowerShellGet 3 Beta 11.

## Screenshots
